### ISSUE TITLE
Filter dropdown should include the `sameDay` and `sameTime` operators

### ISSUE THREAD

**Issue by ExtAnimal** [bryntum/support#10373]
[Forum post](https://forum.bryntum.com/viewtopic.php?p=158473#p158473)

https://bryntum.com/products/gantt/examples/advanced

Add "lag" of 9 hours to one of the deeply nested tasks. It will then start at 9am on what is reported as its `startDate`

The poster was then selecting the `equals` option in the column header filter dropdown, and the date will not match the date selected, because the date selected is at time `00:00:00`.

The `CollectionFilter` class has the `sameDay` and `sameTime` operator. These should be made available in the filter feature.

**Comment by ExtAnimal** [comment]
The online example can be tweaked to include filtering using a conditional breakpoint in `shared.js`:

![Screenshot 2024-11-20 at 11 47 36](https://github.com/user-attachments/assets/842aab9b-5882-497c-9f77-44f399bda029)

**Comment by ExtAnimal** [comment]
Also, we need to make the PHP script which does the conversion reject HTTP requests which are not from the example page.

People should not use our example server for their apps.

**Comment by isglass** [comment]
I agree that date filtering using the UI should probably be more "forgiving", `sameDay` / `notSameDay` sounds good to me. While maybe not as useful, I can still see use for `sameTime` too
